src/types/types.ts

```diff
@@ -713,15 +718,30 @@ export type DeleteAssetMapping3D = AssetMapping3DBase;
 export type EXECUTE = 'EXECUTE';
 
 /**
- * Query schema for asset aggregate endpoint
+ * Query schema for event aggregate endpoint
  */
-export interface EventAggregateQuery {
+export type EventAggregateQuery = EventCountAggregate | EventAdvancedAggregate;
+
+export interface EventBaseAggregate {
   /**
    * Filter on events with strict matching.
    */
   filter?: EventFilter;
 }
 
+export interface EventCountAggregate extends EventBaseAggregate {}
+
+export interface EventAdvancedAggregate extends EventBaseAggregate {
+  /**
+   * The field name(s) to apply the aggregation on. Currently limited to one field.
+   */
+  fields: ('type' | 'subtype' | 'dataSetId')[];
```

Comment:
can we add a code snippet in the `eventsApi.ts:aggregate(...)`
even I don't understand how it works from the first glance

Reply:
@polomani the plan is to have a function per aggregate type (so a single API can spawn multiple SDK functions). Not sure how snippets would work there

Reply:
whose plain is that?
so far we had one function per endpoint why do you want to change that?
this will introduce a huge gap between api schemas and typescript types
what are the reasons?

Reply:
ok